pilot/pkg/xds/discovery.go

```diff
@@ -431,15 +431,31 @@ func configsUpdated(req *model.PushRequest) string {
 }
 
 func reasonsUpdated(req *model.PushRequest) string {
+	var (
+		reason0, reason1            model.TriggerReason
+		reason0Cnt, reason1Cnt, idx int
+	)
+	for r, cnt := range req.Reason {
+		if idx == 0 {
+			reason0, reason0Cnt = r, cnt
+		} else if idx == 1 {
+			reason1, reason1Cnt = r, cnt
+		} else {
+			break
+		}
+		idx++
+	}
+
 	switch len(req.Reason) {
 	case 0:
 		return "unknown"
 	case 1:
-		return string(req.Reason[0])
+		return fmt.Sprintf("%s:%d", reason0, reason0Cnt)
 	case 2:
-		return fmt.Sprintf("%s and %s", req.Reason[0], req.Reason[1])
+		return fmt.Sprintf("%s:%d and %s:%d", reason0, reason0Cnt, reason1, reason1Cnt)
 	default:
-		return fmt.Sprintf("%s and %d more reasons", req.Reason[0], len(req.Reason)-1)
+		return fmt.Sprintf("%s:%d and %d(%d) more reasons", reason0, reason0Cnt, len(req.Reason)-1,
+			req.Reason.Count()-reason0Cnt)
```

Comment:
Do we really care about the count, if not can we remove

Reply:
I think these data are meaningful in practice, making it easy to observe the number of changes triggered in the system to evaluate the system's status.

Reply:
There is metric recorded in ConfigUpdate

Reply:
Logs and metrics are two ways of observation, and in troubleshooting, logs have better real-time performance and contextual relevance. Based on this, I tend to let logs convey more information.